crates/router/src/core/payment_methods/network_tokenization.rs

```
@@ -308,18 +314,81 @@ pub async fn make_card_network_tokenization_request(
     }
 }
 
+#[cfg(all(feature = "v2", feature = "payment_methods_v2"))]  
+pub async fn make_card_network_tokenization_request(
+    state: &routes::SessionState,
+    card: &api_payment_methods::CardDetail,
+    customer_id: &id_type::GlobalCustomerId,
+) -> CustomResult<
+    (NetworkTokenDetails, String),
+    errors::NetworkTokenizationError,
+> {
+
+    let card_data = domain::CardData {
+        card_number: card.card_number.clone(),
+        exp_month: card.card_exp_month.clone(),
+        exp_year: card.card_exp_year.clone(),
+        card_security_code: None,
+    };
+
+    let payload = card_data
+        .encode_to_string_of_json()
+        .and_then(|x| x.encode_to_string_of_json())
+        .change_context(errors::NetworkTokenizationError::RequestEncodingFailed)?;
+
+    let payload_bytes = payload.as_bytes();
+    let (resp, network_token_req_ref_id) = if let Some(network_tokenization_service) = &state.conf.network_tokenization_service {
+        record_operation_time(
+            async {
+                mk_tokenization_req(
+                    state,
+                    payload_bytes,
+                    customer_id.clone(),
+                    network_tokenization_service.get_inner(),
+                )
+                .await
+                .inspect_err(
+                    |e| logger::error!(error=?e, "Error while making tokenization request"),
+                )
+            },
+            &metrics::GENERATE_NETWORK_TOKEN_TIME,
+            router_env::metric_attributes!(("locker", "rust")),
+        )
+        .await
+    } else {
+        Err(errors::NetworkTokenizationError::NetworkTokenizationServiceNotConfigured)
+            .inspect_err(|_| {
```

Why here discarding error?